util/runner.zig

```diff
@@ -47,30 +47,42 @@ const State = union(enum) {
 
         /// Array of timings collected.
         timings_ns: []u64,
+
+        /// Array of maximum memory allocation readings.
+        allocation_maxes: ?[]usize,
+
+        /// Array of counts of memory allocation readings.
+        allocation_counts: ?[]usize,
     };
 };
 
 allocator: std.mem.Allocator,
+track_allocations: bool,
 state: State,
 
 pub fn init(
     allocator: std.mem.Allocator,
     iterations: u16,
     max_iterations: u16,
     time_budget_ns: u64,
+    track_allocations: bool,
 ) Error!Runner {
     return if (iterations == 0) .{
         .allocator = allocator,
+        .track_allocations = track_allocations,
         .state = .{ .preparing = .{
             .max_iterations = max_iterations,
             .time_budget_ns = time_budget_ns,
         } },
     } else .{
         .allocator = allocator,
+        .track_allocations = track_allocations,
         .state = .{ .running = .{
             .iterations_count = iterations,
             .iterations_remaining = iterations,
             .timings_ns = try allocator.alloc(u64, iterations),
+            .allocation_maxes = if (track_allocations) try allocator.alloc(usize, iterations) else null,
```

Comment:
What do you think of encapsulating the tracking logic within a dedicated struct?
It could encapsulate the tracking logic in a single place, making it easier to manage and modify.

Something like this?
```zig
const AllocationTracker = struct {
    maxes: []usize,
    counts: []usize,

    pub fn init(allocator: *std.mem.Allocator, iterations: usize, track: bool) !?AllocationTracker {
        if (!track) return null;
        return AllocationTracker{
            .maxes = try allocator.alloc(usize, iterations),
            .counts = try allocator.alloc(usize, iterations),
        };
    }

    pub fn update(self: *AllocationTracker, index: usize, max: ?usize, count: ?usize) void {
        if (max) |m| self.maxes[index] = m;
        if (count) |c| self.counts[index] = c;
    }

    pub fn deinit(self: *AllocationTracker, allocator: *std.mem.Allocator) void {
        allocator.free(self.maxes);
        allocator.free(self.counts);
    }
};
```

Reply:
That's a good idea I think, then the maxes and counts are always null or not null in synch.